### DISCUSSION TITLE
Newbie help: single node mode and MaxMind

### DISCUSSION THREAD

**Discussion by beneix** [pfelk/pfelk#455] [Q&A]
I am trying to install pfELK on a machine on my LAN (following the [how-to for docker-compose](https://github.com/pfelk/docker) and have a couple of questions:

1. My machine has modest hardware so I'd like to maximise performance. I thought that it would be a good idea to run pfELK on a single-node setup, so I wanted to modify the docker-compose.yml file accordingly. The only instruction I found on Github was to [modify /etc/elasticsearch/elasticsearch.yml](https://github.com/pfelk/pfelk/wiki/How-To:-Performance#elasticsearch-single-node), but a) that file does not exist before you start the install and b) I would have thought that docker-compose.yml also needs to be modified. Do I need to change the _create certs_ and _environment_ sections, and if so how? Alternatively, if running three nodes does not consume more resources than a single node, please let me know.

2. Also, I'd like to set up MaxMind, and I'd like to do it on Docker since my machine is running Alpine Linux and I don't think there is a repository for MaxMind available. I have found a [Docker container](https://hub.docker.com/r/observabilitystack/geoip-api#!) for the purpose, but I am not sure exactly how pfElk speaks to MaxMind so I need some more info to make sure the two can communicate. The [pfELK how-to for MaxMind](https://github.com/pfelk/pfelk/wiki/How-To:-MaxMind-via-GeoIP-with-pfELK) does not mention the required interface with MaxMind so I don't know what the prerequisite is when not installing MaxMind in the standard way.

**Comment by a3ilson** [comment]
@beneix - any luck?

I'm working to refine and update the current docker-compose. pfELK reads the MaxMind databases via file path...might be best to download and create docker volume,
